packages/interactivity/src/store.ts

```diff
@@ -125,7 +125,12 @@ const handlers = {
 					try {
 						value = await it.value;
 					} catch ( e ) {
+						setNamespace( ns );
+						setScope( scope );
 						gen.throw( e );
+					} finally {
+						resetScope();
+						resetNamespace();
```

Comment:
@DAreRodz I tried 2 different approaches and they both fixed my test case. I wasn't sure why we have 2 try blocks, so my first version merge the try/finally and try/catch into one try/catch/finally. Then, since I wasn't sure what the motivation for 2 blocks were, I changed use 2 blocks. I'd like to get your insight here.

Both versions are on this branch in different commits and can be reviewed.

Reply:
I think this is the best approach: to have two separate try-catch blocks. The thing with async actions is that they can change the scope at any time―whenever they run―, thus leading to unexpected results when the scope is reset.

We decided here to set/reset the scope for synchronous code fragments, i.e., the generator code from one `yield` to the next one. So, from my understanding, we should reset the scope, then wait for the promise to be fulfilled (which will probably set their own scope), and if it fails, then set/reset the scope again.